cmake/HPX_SetupApex.cmake

```diff
@@ -53,6 +53,10 @@ if(HPX_WITH_APEX AND NOT TARGET APEX::apex)
     target_link_libraries(APEX::apex INTERFACE APEX::apex)
   else()
     target_link_libraries(APEX::apex INTERFACE apex)
+    if(TARGET Mpi::mpi)
+      target_compile_definitions(apex PRIVATE APEX_HAVE_MPI)
+      target_link_libraries(APEX::apex PRIVATE Mpi::mpi)
+    endif()
```

Comment:
Does APEX not set up this dependency by itself? We might still need this here because APEX being special... It looks like APEX has a CMake variable called `USE_MPI` which we could set instead but I don't know if that's intended for public use. @khuck?

Reply:
I left this in as the original PR was handling this in a special way. I have no means of verifying it, however.

Reply:
@msimberg APEX doesn't depend on MPI support when used with HPX.  In fact, in my head they are mutually exclusive... why does APEX require MPI for this PR?

Reply:
@khuck no idea. @biddisco had that in his initial version for this PR. I'll go ahead and remove this, then.

Reply:
I don't know the original reason either, but does APEX require MPI to be on to correctly handle multiple localities? I think the idea is that you can have the module introduced in this PR enabled with MPI on, but HPX as would have networking off (and thus no MPI parcelport). In that case HPX has no intrinsic concept of "localities" but would instead verbatim use MPI ranks to identify different processes. If APEX handles that no matter if MPI is enabled or not I think this can be removed, but only @biddisco knows for sure.

Reply:
I have removed it for now. If things break for @biddisco after that we can always re-introduce this later. I'd rather have this PR in master asap than to wait.